lib/utils/docker.coffee

```diff
@@ -244,3 +290,65 @@ exports.getDocker = (options) ->
 exports.getImageSize = (docker, image) ->
 	docker.getImage(image).inspectAsync()
 	.get('Size')
+
+hasQemu = ->
+	fs = require('mz/fs')
+
+	getQemuPath()
+	.then (qemu) ->
+		fs.stat(qemu)
+	.return(true)
+	.catchReturn(false)
+
+getQemuPath = ->
+	resin = require('resin-sdk-preconfigured')
+	path = require('path')
+	fs = require('mz/fs')
+
+	resin.settings.get('binDirectory')
+	.then (binDir) ->
+		# The directory might not be created already,
+		# if not, create it
+		fs.access(binDir)
+		.catch code: 'ENOENT', ->
+			fs.mkdir(binDir)
+		.then ->
+			path.join(binDir, 'qemu-execve')
+
+platformNeedsQemu = ->
+	os = require('os')
+	os.platform() == 'linux'
+
+installQemu = ->
+	request = require('request')
+	fs = require('fs')
+	zlib = require('zlib')
+
+	qemuVersion = 'v2.5.50-resin-execve'
+
+	getQemuPath()
+	.then (qemuPath) ->
+		installStream = fs.createWriteStream(qemuPath)
+		qemuUrl = "https://github.com/resin-io/qemu/releases/download/#{qemuVersion}/qemu-execve.gz"
+		request(qemuUrl)
+		.pipe(zlib.createGunzip())
+		.pipe(installStream)
```

Comment:
We need it to definitely be there when the build begins, as it gets copied into the build context. I feel like this is the safest method to ensure the binary will be present and fully downloaded.